### ISSUE TITLE
Error opening Database.  You probably need to disable SIP for this to work.

### ISSUE THREAD

**Issue by edglobo-hmedeiros** [jacobsalmela/tccutil#62]
Trying to execute some tccuilts commands, but no one is succesfully.

Already set full disk control for terminal app. And the issue remains.
What i need to do?

```
medeiroshudson ~ % sudo tccutil --digest
/opt/homebrew/bin/tccutil:29: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  osx_version = version(mac_ver()[0])  # mac_ver() returns 10.16 for Big Sur instead 11.+
Error opening Database.  You probably need to disable SIP for this to work.
```

Thanks.

**Comment by jacobsalmela** [comment]
What version of `tccutil.py` do you have? Have you looked at this thread?

https://github.com/jacobsalmela/tccutil/discussions/44

I still have to validate some things regarding it.

**Comment by fengyuan213** [comment]
not working on macOs Venutura13.2.1 (22D68), I already turned the sip off.

```/usr/local/bin/tccutil:29: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
  osx_version = version(mac_ver()[0])  # mac_ver() returns 10.16 for Big Sur instead 11.+
Error opening Database.  You probably need to disable SIP for this to work.```

**Comment by jacobsalmela** [comment]
I suspect this will be less and less likely to work with each release.  We can use this issue to figure it out for all versions, but I cannot get to the project for a while...

**Comment by probablypablito** [comment]
Any update on this? Having the same issue as root & SIP off on Ventura 13.1

**Comment by ylluminate** [comment]
Now with the update going into Homebrew and seeing the notification, this has come back to the front of my mind. Please let us know if there's anything further anyone of us might do to help @jacobsalmela!

**Comment by jacobsalmela** [comment]
Yes, I need help.  Lol.

I don't have valid test rigs anymore for all the OS flavors and the DB has changed.

I do think Apple will block things like this more actively and since I haven't made many of the updates as of late, I am not sure where to start diagnosing and making a solution.  Let alone the time to do it amongst my other projects and busy home life

**Comment by ylluminate** [comment]
Understand that!!!!!!! If you can offer some steps / suggestions / info you may need and how we can facilitate it, would be happy to gather info and so on.

**Comment by jacobsalmela** [comment]
Awesome!

I think we need to find the last version of macOS this worked with reliably.  I think it may even have been the last big cat names or 1/2 releases after those... Then we can narrow down the change.

I think we would also need a dump of the DB schema for the required table(s).  And verify who can/can't write to it with and without SIP enabled.

Something like that would probably be a good start.

**Comment by ylluminate** [comment]
Hmm - last "[big cat name](https://wikiless.tiekoetter.com/wiki/List_of_Apple_codenames?lang=en)" was 10.8 (Mountain Lion), but I didn't think TCC came around until 10.14'ish... Am I misunderstanding?

Would be happy to do some data gathering here once we nail this down.

**Comment by jacobsalmela** [comment]
Obviously, I can't keep track of macOS releases and time is skewed in my brain.  

Guess we could just look at the first version I supported for the app 😃

**Comment by ylluminate** [comment]
So my thought initially, then, is to attempt to test on 10.14 and then test on 10.15 and report back hopefully any pertinent information. I am not certain if I can do it sans SIP given that these will be installs via Parallels on an old MacPro I have and I recall that some things were a bit dicey from time to time.
